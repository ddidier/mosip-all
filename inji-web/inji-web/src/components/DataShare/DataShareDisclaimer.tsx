import React from "react";
import {DSDisclaimerProps} from "../../types/components";

export const DataShareDisclaimer:React.FC<DSDisclaimerProps> = (props) => {
    return <div className={"flex w-full p-2 rounded-lg border-2 border-iw-shieldLoadingShadow bg-iw-disclaimerBackGround my-8"} data-testid={"DataShareDisclaimer-Outer-Container"}>
        <div className={"px-4 text-center text-iw-disclaimerText text-xs"} data-testid={"DataShareDisclaimer-Content"}>{props.content}</div>
    </div>;
}

