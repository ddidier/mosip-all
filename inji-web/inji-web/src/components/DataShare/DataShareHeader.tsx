import React from "react";
import {DSHeaderProps} from "../../types/components";

export const DataShareHeader:React.FC<DSHeaderProps> = (props) => {
    return <div className="flex items-start justify-between p-5 border-b border-solid rounded-t" data-testid={"DataShareHeader-Outer-Container"}>
        <div className="flex flex-col">
            <div className="text-lg font-semibold"  data-testid={"DataShareHeader-Header-Title"}> {props.title}</div>
            <div className="text-md text-iw-subTitle"  data-testid={"DataShareHeader-Header-SubTitle"}> {props.subTitle} </div>
        </div>
    </div>;
}

