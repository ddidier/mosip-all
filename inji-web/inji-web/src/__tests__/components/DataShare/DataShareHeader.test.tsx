import {screen} from "@testing-library/react";
import {DataShareHeader} from "../../../components/DataShare/DataShareHeader";
import { renderWithProvider } from "../../../test-utils/mockUtils";

describe("Testing Layout of the Expiry Header", () => {
    
    test("Check if the layout is matching with the snapshots", ()=>{
        const{asFragment} = renderWithProvider(<DataShareHeader title={"title"} subTitle={"subTitle"}/>)
        expect(asFragment()).toMatchSnapshot();
    })
});
describe("Testing Functionality of the Expiry Header",() =>{
    test("Check Presence of the Header Title", ()=>{
        renderWithProvider(<DataShareHeader title={"title"} subTitle={"subTitle"}/>);
        expect(screen.getByTestId("DataShareHeader-Header-Title")).toHaveTextContent("title");
    })
    test("Check Presence of the Header SubTitle", ()=>{
        renderWithProvider(<DataShareHeader title={"title"} subTitle={"subTitle"}/>);
        expect(screen.getByTestId("DataShareHeader-Header-SubTitle")).toHaveTextContent("subTitle");
    })
});
