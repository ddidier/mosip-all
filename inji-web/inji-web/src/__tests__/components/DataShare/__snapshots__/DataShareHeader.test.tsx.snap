// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Layout of the Expiry Header Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    class="flex items-start justify-between p-5 border-b border-solid rounded-t"
    data-testid="DataShareHeader-Outer-Container"
  >
    <div
      class="flex flex-col"
    >
      <div
        class="text-lg font-semibold"
        data-testid="DataShareHeader-Header-Title"
      >
         title
      </div>
      <div
        class="text-md text-iw-subTitle"
        data-testid="DataShareHeader-Header-SubTitle"
      >
         subTitle 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
